### PR TITLE
Different fixes in the asyncapi v3 spec examples

### PR DESCRIPTION
**Pull request by ctasada**
Some of the used examples to test the code were not passing the AsyncAPI v3 validation process. Here we fix them and leave the code ready for a next iteration of improvements

### CODE REVIEW

File: springwolf-plugins/springwolf-sns-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/annotation/SnsAsyncOperationBinding.java [hunk 1]
```diff
@@ -20,4 +20,10 @@
 public @interface SnsAsyncOperationBinding {
```
Comment (ctasada) [inline springwolf-plugins/springwolf-sns-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/annotation/SnsAsyncOperationBinding.java:20]: Both `protocol` and `endpoint` are mandatory fields for which I don't think we should provide a default value. This decision automatically generates a breaking change. Without a value, the compilation of any code using this annotation will fail.

File: springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/annotation/AsyncAnnotationChannelsScanner.java [hunk 2]
```diff
@@ -94,13 +94,7 @@ private Stream<MethodAndAnnotation<A>> getAnnotatedMethods(Class<?> type) {
...
 
     private boolean isInvalidChannel(Map.Entry<String, ChannelObject> entry) {
-        //        Operation publish = entry.getValue().getPublish();
-        //        boolean publishBindingExists = publish != null && publish.getBindings() != null;
-        //
-        //        Operation subscribe = entry.getValue().getSubscribe();
-        //        boolean subscribeBindingExists = subscribe != null && subscribe.getBindings() != null;
-
-        boolean allNonNull = entry.getKey() != null; // && (publishBindingExists || subscribeBindingExists); FIXME
+        boolean allNonNull = entry.getKey() != null;
```
Comment (timonback) [inline springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/annotation/AsyncAnnotationChannelsScanner.java:97]: I believe a map key can never be null be design.
-> remove this method?

File: springwolf-plugins/springwolf-cloud-stream-plugin/src/test/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/cloudstream/CloudStreamFunctionChannelsScannerIntegrationTest.java [hunk 3]
```diff
@@ -98,30 +101,34 @@ void testConsumerBinding() {
...
 
         ChannelObject expectedChannel = ChannelObject.builder()
                 .bindings(channelBinding)
-                // .publish(operation) FIXME
+                .messages(Map.of(message.getName(), message))
                 .build();
 
-        assertThat(channels).containsExactly(Map.entry(topicName, expectedChannel));
+        Operation expectedOperation = Operation.builder()
+                .action(OperationAction.SEND)
```
Comment (timonback) [inline springwolf-plugins/springwolf-cloud-stream-plugin/src/test/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/cloudstream/CloudStreamFunctionChannelsScannerIntegrationTest.java:122]: Nitpicking: Receive action

**Review by timonback — APPROVED** [review]
Again, great work!

So good to see an actual sqs/sns binding

File: springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/annotation/AsyncAnnotationChannelsScanner.java (hunk 2 above)
Comment (ctasada) [reply springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/annotation/AsyncAnnotationChannelsScanner.java:97]: Yes, I think that would be the best. If in the future we find out that some conditions are generating broken instances of Channel or Operation, we can add back the filter

File: springwolf-plugins/springwolf-cloud-stream-plugin/src/test/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/cloudstream/CloudStreamFunctionChannelsScannerIntegrationTest.java (hunk 3 above)
Comment (ctasada) [reply springwolf-plugins/springwolf-cloud-stream-plugin/src/test/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/cloudstream/CloudStreamFunctionChannelsScannerIntegrationTest.java:122]: I will try to fix it tonight

**Review by timonback — APPROVED** [review]

**Pull request merged**